pkgs/applications/video/obs-studio/plugins/obs-vkcapture.nix

```diff
@@ -22,8 +23,27 @@ stdenv.mkDerivation rec {
     hash = "sha256-yaN0am24p9gC+s64Rop+jQ3952UOtZund/KttnVxP48=";
   };
 
+  cmakeFlags = lib.optionals stdenv.isi686 [
+    "-DBUILD_PLUGIN=OFF"
+  ];
+
   nativeBuildInputs = [ cmake ninja ];
-  buildInputs = [ libGL libX11 obs-studio vulkan-headers vulkan-loader wayland ];
+  buildInputs = [
+    libGL
+    libX11
+    vulkan-headers
+    vulkan-loader
+    wayland
+  ]
+  ++ lib.optionals (!stdenv.isi686) [
+    obs-studio
+  ];
+
+  # Support 32bit Vulkan applications by linking in the 32bit Vulkan layer
+  postInstall = lib.optionalString (stdenv.hostPlatform.system == "x86_64-linux") ''
```

Comment:
We can't get this file without building the package for 32bit including all dependencies? Would cp'ing this file reduce the dependency tree? If we compare to ``"x86_64-linux"`` we should restrict platforms to ``"x86_64-linux" "i686-linux``

Reply:
> We can't get this file without building the package for 32bit including all dependencies?

We could, it's just text. That wouldn't help us though and I'm pretty sure nix would fail build.

> Would cp'ing this file reduce the dependency tree?

No. The file references the 32bit build's .so that is to be loaded by VK.

> If we compare to `"x86_64-linux"` we should restrict platforms to `"x86_64-linux" "i686-linux`

I don't see a reason why aarch64-linux or riscv-linux shouldn't have access to obs-vkcapture to capture native VK games.

On x86_64, we have the special case that we want the i686-linux layer aswell. When the time comes where we need/want the armv7l-linux layer in addition to the aarch64-linux one, it'd be simple enough to change.

This is the same pattern employed by Mangohud and similar VK layers FYI.